attributes/attributes.go

```diff
@@ -66,5 +69,22 @@ func (a *Attributes) WithValues(kvs ...interface{}) *Attributes {
 // Value returns the value associated with these attributes for key, or nil if
 // no value is associated with key.
 func (a *Attributes) Value(key interface{}) interface{} {
+	if a == nil {
+		return nil
+	}
 	return a.m[key]
 }
+
+// Clone returns a new Attributes containing all key/value pairs found in a.
+// Since Attributes stores its key/value paris as type `interface{}`, it is not
+// possible to perform deep copies here.
+func (a *Attributes) Clone() *Attributes {
```

Comment:
Attributes are immutable, so if we can't do a deep copy, I don't think there's any value of a `Clone()`, is there?

Reply:
`WithValues()` is already making a clone.

Reply:
I guess what I wanted to say was that since `Attributes` stores key/value pairs of type `interface{}`, one can store a pointer as the value type. The caller of the `Clone` operation can then modify the contents of the data pointed to by the value. This is no different from what `WithValues` does. So, maybe I should remove that comment about the deep copy.

In fact I could and maybe i should replace the implementation of `Clone` as follows:
```golang
func (a *Attributes) Clone() *Attributes {
    return a.WithValues()
}
```

Reply:
But why have `Clone` at all?  Or call `Attributes.WithValues` with no parameters?  There is no way to modify a `*Attributes`, so there is no need to `Clone` it.  I.e.:

```
a := attributes.New(kvs...)
b := a.WithValues(otherKVs...)
c := a.WithValues(otherOtherKVs...)
```

`a`, `b`, and `c` will all have `kvs`, `b` will be the only thing with `otherKVs`, and `c` will be the only one with `otherOtherKVs`.  You can safely pass around and make copies of `*Attributes`.

If a user stores a pointer in `*Attributes` and modifies it later, `Clone`-ing the `*Attributes` won't change the fact that modification to that pointer will modify the `*Attributes` (clones and the original).  The only way to prevent that would have been to force the values stored in `Attributes` to have a `Clone` method on them, and call that in `WithValues`/`Clone`.

Reply:
Makes sense, got rid of `Clone`.